mysite/ct/views.py

```diff
@@ -155,8 +155,8 @@ def fsm_redirect(self, request, eventName=None, defaultURL=None,
         # now we check here whether event is actually from path matching
         # this node
         referer = request.META.get('HTTP_REFERER', '')
-        referer = referer[referer.find('/ct/'):]
-        if request.method == 'POST' and referer != self.fsmStack.state.path \
+        origin = request.META.get('HTTP_ORIGIN', '')
```

Comment:
@maxsocl Sorry, but my browser (Firefox 38) is not passing HTTP_ORIGIN header, so this crashes with 500 Server Error.  I haven't tested other browsers yet.  Running a quick google search, I'm seeing people talking about ORIGIN only being for cross-site validation and not necessarily getting sent by all user agents all the time... so I think our code must handle the situation where it's missing... as in the very first time I tested this.